pandas/core/categorical.py

```diff
@@ -213,13 +207,15 @@ def __init__(self, values, categories=None, ordered=False, name=None, fastpath=F
         if fastpath:
             # fast path
             self._codes = _coerce_indexer_dtype(values, categories)
-            self.name = name
             self.categories = categories
             self._ordered = ordered
             return
 
-        if name is None:
-            name = getattr(values, 'name', None)
+        if not name is None:
+            msg = "the 'name' keyword is removed, use 'name' with consumers of the " \
+                  "categorical instead (e.g. 'Series(cat, name=\"something\")'"
+            import warnings
+            warnings.warn(msg, FutureWarning)
```

Comment:
can you add `stacklevel=2` here? (to ensure that towards the user the warning is raised for the code he called on not this line)